cmd/test-server/main.go

```diff
@@ -178,48 +196,164 @@ func pingHandler(w http.ResponseWriter, r *http.Request) {
+				select {
+				case <-ticker.C:
+					tResult = append(tResult, sendTcpPing(dst, seqNumber, TCPTimeout))
+				}
+			}
+			ticker.Stop()
+			min, avg, max, stddev := getStats(tResult)
+			tcpResultArr = append(tcpResultArr, tcpResult{port, tResult, min, avg, max, stddev})
+		}(port)
+	}
+	portsWaitGroup.Wait()
+	tcpResultObj := tcpStruct{ip, tcpResultArr}
+	return tcpResultObj
+}
+
+// Avg RTT from all successful ICMP measurements, to display on webpage
+func getAvgIcmpResult(icmp []RtItem) float64 {
+	var sum float64 = 0
+	var len float64 = 0
+	for _, x := range icmp {
+		if x.AvgRtt == 0 {
+			continue
+		}
+		sum += x.AvgRtt
+		len += 1
+	}
+	var avg float64 = sum / len
+	return avg
+}
+
+// Avg RTT from all successful TCP measurements regardless of port, to display on webpage
+func getAvgTcpResult(tcp []tcpStruct) float64 {
+	var sum float64 = 0
+	var len float64 = 0
+	// for each IP in subnet
+	for _, x := range tcp {
+		// for each port per IP
+		for _, p := range x.Probes {
+			if p.AvgRtt == 0 {
+				continue
 			}
+			sum += p.AvgRtt
+			len += 1
+		}
+	}
+	var avg float64 = sum / len
+	return avg
+}
+
+// Handler for ICMP and TCP measurements which also serves the webpage via a template
+func pingHandler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/ping" {
+		http.NotFound(w, r)
+		return
+	}
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusNotImplemented)
+		w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+		return
+	}
+	clientIPstr := r.RemoteAddr
+	clientIP, _, _ := net.SplitHostPort(clientIPstr)
+	adjIPstoPing := getAdjacentIPs(clientIP)
+	var expUUID = uuid.NewString()
+	var timestamp = time.Now().Format("2006-01-02T15:04:05.000000") //RFC3339 style date with added seconds information
+	ipTotal := len(adjIPstoPing)
+	offset := 0
+	numBatches := int(math.Ceil(float64(ipTotal / batchSizeLimit)))
+	var icmpResults []RtItem
+	var tcpResultsObj []tcpStruct
+	for i := 0; i <= numBatches; i++ {
+		lower := offset
+		upper := offset + batchSizeLimit
+
+		if upper > ipTotal {
+			upper = ipTotal
 		}
-		ticker.Stop()
-		min, avg, max, stddev := getStats(tResult)
-		tcpResultArr = append(tcpResultArr, tcpResult{dst, tResult, min, avg, max, stddev})
+		batchIPs := adjIPstoPing[lower:upper]
+		offset += batchSizeLimit
+
+		var icmpWaitGroup sync.WaitGroup
+		var tcpWaitGroup sync.WaitGroup
+
+		icmpWaitGroup.Add(len(batchIPs))
+		tcpWaitGroup.Add(len(batchIPs))
+
+		for id := range batchIPs {
+			go func(IP string, id int) {
+				defer icmpWaitGroup.Done()
+				icmpResults = append(icmpResults, IcmpPinger(IP))
+			}(batchIPs[id], id)
+			go func(IP string, id int) {
+				defer tcpWaitGroup.Done()
+				tcpResultsObj = append(tcpResultsObj, TcpPinger(IP))
+			}(batchIPs[id], id)
+		}
+		icmpWaitGroup.Wait()
+		tcpWaitGroup.Wait()
 	}
 
 	// Combine all results
 	results := Results{
-		UUID:     expUUID,
-		IPaddr:   clientIP,
-		IcmpPing: icmp,
-		TcpPing:  tcpResultArr}
+		UUID:        expUUID,
```

Comment:
To avoid the earlier variable declaration. The fewer variables the reader needs to keep track off, the easier is it to read the code.
```suggestion
		UUID:        uuid.NewString(),
```